Otus-DevOps-2018-02/ifqthenp_infra
# Homework 6. Packer base

# Completed Homework №6


## The following tasks have been completed

- :large_blue_diamond: Created `packer` template file `ubuntu16.json` and built `reddit-base` image with it
- :large_blue_diamond: Created new VM instance using `reddit-base` image and deployed reddit app to that instance
- :large_blue_diamond: Parameterized `ubuntu16.json` template file with `variables.json`
- :large_blue_diamond: Added extra options for `packer` builder in `variable.json` as requested in Homework 6
- :large_orange_diamond: Created `immutable.json` template file and built `reddit-full` image with it
- :large_orange_diamond: Used provisioners in `immutable.json` to ensure that new VM instance created with `reddit-full` image runs fully working reddit application upon deployment
- :large_orange_diamond: Added new script `create-reddit-vm.sh` to `config-scripts` folder to speed up VM instance creation process

## How to get project running

- To build new image run `packer build immutable.json` from `packer` directory
- To create VM instance using newly built image run `./create-reddit-vm.sh` from `config-scripts` directory

## How to check that project actually works

- Ensure that the build passed TravisCI tests either
  - in [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - or on actual GitHub pull request page
- Upon deployment ensure that application returns `200 OK` status code. Run `curl -s -I 35.197.206.39:9292`
- Open in browser http://35.197.206.39:9292

## PR checklisk
